x-pack/plugins/lens/public/xy_visualization_plugin/xy_visualization.tsx

```diff
@@ -7,69 +7,49 @@
 import React from 'react';
 import { render } from 'react-dom';
 import { Position } from '@elastic/charts';
-import { Visualization, Operation } from '../types';
+import uuid from 'uuid';
+import { I18nProvider } from '@kbn/i18n/react';
 import { getSuggestions } from './xy_suggestions';
-import { XYArgs } from './xy_expression';
-import { NativeRenderer } from '../native_renderer';
-
-export type State = XYArgs;
-export type PersistableState = XYArgs;
+import { XYConfigPanel } from './xy_config_panel';
+import { Visualization } from '../types';
+import { State, PersistableState } from './types';
 
 export const xyVisualization: Visualization<State, PersistableState> = {
   getSuggestions,
 
   initialize(state) {
     return (
       state || {
-        title: 'Empty line chart',
-        legend: { isVisible: true, position: Position.Right },
         seriesType: 'line',
-        splitSeriesAccessors: [],
-        stackAccessors: [],
+        title: 'Empty XY Chart',
+        legend: { isVisible: true, position: Position.Right },
         x: {
-          accessor: '',
+          accessor: uuid.v4(),
```

Comment:
Maybe I misunderstood the intention, but shouldn't the datasource decide the column id?

Reply:
Ah. Nope. You understood perfectly. Originally, this responsibility belonged to the visualization, but then we decided to move it into the data-source. This is vestigial....